delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/PurgeElemFailures.scala

```diff
@@ -1,29 +1,28 @@
 package ch.epfl.bluebrain.nexus.delta.sourcing
 
-import cats.effect.{Clock, IO}
+import cats.effect.IO
 import ch.epfl.bluebrain.nexus.delta.kernel.Logger
 import ch.epfl.bluebrain.nexus.delta.sourcing.PurgeElemFailures.logger
-import ch.epfl.bluebrain.nexus.delta.sourcing.config.ProjectionConfig
-import ch.epfl.bluebrain.nexus.delta.sourcing.stream.{CompiledProjection, ExecutionStrategy, ProjectionMetadata, Supervisor}
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.PurgeConfig
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.PurgeProjectionCoordinator.PurgeProjection
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.ProjectionMetadata
 import doobie.implicits._
 import doobie.postgres.implicits._
-import fs2.Stream
 
-import scala.concurrent.duration._
+import java.time.Instant
 
-final class PurgeElemFailures private[sourcing] (xas: Transactors, ttl: FiniteDuration, clock: Clock[IO]) {
+final class PurgeElemFailures private[sourcing] (xas: Transactors) {
 
   /**
-    * Deletes the projection errors that are older than the given `ttl`.
+    * Deletes the projection errors that are older than the given instant.
     */
-  def apply(): IO[Unit] =
+  def apply(instant: Instant): IO[Unit] =
```

Comment:
The threshold instant is already computed from the info from the `PurgeProjection`